src/host/screenInfo.cpp

```diff
@@ -2464,6 +2469,12 @@ std::pair<COORD, COORD> SCREEN_INFORMATION::GetWordBoundary(const COORD position
             }
         }
     }
+
+    // The calculated range is in buffer coordinates, but the caller is
+    // expecting screen offsets, so we have to convert these back again.

```

Comment:
This is thanks to you adding a `false` (for useBufferCoordinates) to the text rect computation in Selection, right?

I couldn't immediately evaluate this as true, but I did look at consumers in Selection of the endpoints that are populated through this function and trust it well enough.

Reply:
The only place `GetWordBoundary` is called is from the window mouse event handler. It is passed the mouse position in screen coordinates, and then expects to receive the calculated word boundary in the same coordinate system. See here:
https://github.com/microsoft/terminal/blob/b1e1c7cdf4efa1ca4e0a495f6b90fd83ca8b5633/src/interactivity/win32/windowio.cpp#L803-L806

Reply:
Right -- and that's fed into Selection, which (as far as I can tell) only uses it for updating the renderer's invalidation range. I couldn't determine if anyone was using the stuff in Selection in buffer coords. :smile:

Reply:
It is a bit confusing. Different parts of the `Selection` class use different coordinate systems. For example the `ColorSelection` methods expect buffer coordinates, and the `GetSelectionRects` method returns buffer coordinates, but the internal anchor positions are stored in screen coordinates. This was a bit of a nightmare to sort out, and although I tried my best to catch as many of these cases as I could, it wouldn't be a huge surprise to me if there were others that I missed.